presto-main/src/main/java/io/prestosql/operator/LookupJoinOperator.java

```diff
@@ -676,18 +677,23 @@ private boolean outerJoinCurrentPosition()
                 return blocked(blocked);
             }
 
-            boolean consumedInput = false;
-            if (needsInput() && inputPage != null) {
+            if (inputPage != null && !consumedInputPage) {
                 addInput(inputPage);
-                consumedInput = true;
+                consumedInputPage = true;
             }
 
             Page outputPage = getOutput();
+
+            boolean needsMoreData = consumedInputPage && needsInput();
+            if (needsMoreData) {
+                consumedInputPage = false;
```

Comment:
once you simplify (per above comments), do you still need `consumedInputPage` as a class state?